framework/include/base/Attributes.h

```diff
@@ -239,6 +239,26 @@ class AttribPreAux : public Attribute
   bool _val = false;
 };
 
+/// TODO: delete this later - it is a temporary hack for dealing with inter-system dependencies
+/// this attrib ensures UOs are uniquely assigned a single group to prevent multiple executions
```

Comment:
Can you explain this  a bit more and reference an issue if possible. I don't understand the reason for it based on this comment.

Reply:
Well, the `# TODO` part is only there because the other two attributes have it. 

The initial problem was that, for some reason, I noticed we were calling CSVReader twice when I ran it with `force_preic = true` and `execute_on = INITIAL`, and the whole thing here is we don't want to call this twice for an invariant csv file and due to an error being thrown from the DelimitedFileReader class.

So there are three `Moose:AuxGroup`s that are referenced in FEProblemBase.C: `PRE_IC`, `PRE_AUX`, and `POST_AUX`. However, currently, there are only two attributes that any UO sets, `AttribPreIC` and `AttribPreAux`. By default, both are false. But, when `AttribPreAux`, in particular, is set to false, this means the UO will be executed with the `POST_AUX` group. In contrast, setting the preic attribute to true does nothing to stop it from executing again along with the `POST_AUX` group. So its sort of like an over-determined problem - the goal is to assign a UO to one and only one of three AuxGroups, but we only provide 2 conditions to control this. It really all comes down to this part of FEProblemBase.C:

```
  if (group == Moose::PRE_IC)
    query.condition<AttribPreIC>(true);
  else if (group == Moose::PRE_AUX)
    query.condition<AttribPreAux>(true);
  else if (group == Moose::POST_AUX)
    query.condition<AttribPreAux>(false);
```

It's clear, here, that the `POST_AUX` group only depends on the `AttribPreAux` condition, and has nothing to do with the `AttribPreIC`. So when I ran CSVReader during preic, it had to, by default, execute again with the post-aux group.

Reply:
@aeslaughter Okay, I think I understand now that you are saying to expand on the comment in the actual code....

Yes, I'll make this comment a bit more clear :satisfied: